sim/mage/arcane_blast.go

```diff
@@ -19,40 +20,81 @@ func (mage *Mage) registerArcaneBlastSpell() {
+
+	additiveDamageAffectedSpells := []*core.Spell{}
+	// Purposefully excluded arcane missiles ticks because we manually disable the arcane blast aura after the final tick
+	affectedSpellCodes := []int32{
+		SpellCode_MageArcaneExplosion, SpellCode_MageArcaneSurge, SpellCode_MageLivingFlame, SpellCode_MageSpellfrostBolt,
+	}
 
 	mage.ArcaneBlastAura = mage.GetOrRegisterAura(core.Aura{
 		Label:     "Arcane Blast Aura",
 		ActionID:  core.ActionID{SpellID: 400573},
 		Duration:  time.Second * 6,
 		MaxStacks: 4,
+		OnInit: func(aura *core.Aura, sim *core.Simulation) {
+			additiveDamageAffectedSpells = core.FilterSlice(
+				core.Flatten([][]*core.Spell{
+					mage.ArcaneExplosion,
+					mage.ArcaneMissilesTickSpell,
+					{mage.ArcaneSurge},
+					{mage.SpellfrostBolt},
+				}),
+				func(spell *core.Spell) bool { return spell != nil },
+			)
+		},
 		OnStacksChange: func(aura *core.Aura, sim *core.Simulation, oldStacks int32, newStacks int32) {
 			aura.Refresh(sim)
 			mage.ArcaneBlast.CostMultiplier = 1.75 * float64(newStacks)
+
+			oldMultiplier := .15 * float64(oldStacks)
+			newMultiplier := .15 * float64(newStacks)
+			core.Each(additiveDamageAffectedSpells, func(spell *core.Spell) {
+				spell.DamageMultiplierAdditive -= oldMultiplier
+				spell.DamageMultiplierAdditive += newMultiplier
+			})
+
+			if hasLivingFlameRune {
+				// Living Flame is the only spell buffed multiplicatively for whatever reason
+				mage.LivingFlame.DamageMultiplier /= 1 + oldMultiplier
+				mage.LivingFlame.DamageMultiplier *= 1 + newMultiplier
+			}
+		},
+		OnCastComplete: func(aura *core.Aura, sim *core.Simulation, spell *core.Spell) {
+			if spell.Flags.Matches(SpellFlagMage) && slices.Contains(affectedSpellCodes, spell.SpellCode) {
+				aura.Deactivate(sim)
+			}
 		},
 	})
 
 	mage.ArcaneBlast = mage.RegisterSpell(core.SpellConfig{
 		ActionID:    core.ActionID{SpellID: 400574},
+		SpellCode:   SpellCode_MageArcaneBlast,
 		SpellSchool: core.SpellSchoolArcane,
 		ProcMask:    core.ProcMaskSpellDamage,
 		Flags:       SpellFlagMage | core.SpellFlagAPL,
 
 		ManaCost: core.ManaCostOptions{
-			BaseCost: 0.07,
+			BaseCost: manaCost,
 		},
 		Cast: core.CastConfig{
 			DefaultCast: core.Cast{
 				GCD:      core.GCDDefault,
-				CastTime: time.Millisecond * 2500,
+				CastTime: castTime,
 			},
 		},
 
-		CritMultiplier:   mage.DefaultHealingCritMultiplier(),
 		DamageMultiplier: 1,
-		ThreatMultiplier: 1 - 0.15*float64(mage.Talents.BurningSoul),
+		CritMultiplier:   mage.MageCritMultiplier(0),
+		ThreatMultiplier: 1,
 
 		ApplyEffects: func(sim *core.Simulation, target *core.Unit, spell *core.Spell) {
-			baseDamage := sim.Roll(baseLowDamage, baseHighDamage) + .714*spell.SpellDamage()
+			baseDamage := sim.Roll(baseLowDamage, baseHighDamage) + spellCoeff*spell.SpellDamage()
 			result := spell.CalcAndDealDamage(sim, target, baseDamage, spell.OutcomeMagicHitAndCrit)
 
 			if result.Landed() {
```

Comment:
You also get the debuff(stack) if the spell resists/misses, it's basically "on cast success". I think the block should just be 
```
mage.ArcaneBlastAura.Activate(sim)
mage.ArcaneBlastAura.AddStack(sim)
```